cfme/containers/image_registry.py

```diff
@@ -1,42 +1,65 @@
+
 from cfme.common import SummaryMixin, Taggable
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import CheckboxTable, toolbar as tb
-from cfme.web_ui.menu import nav
+from cfme.web_ui import CheckboxTable, toolbar as tb, paginator, browser_title, controller_name
+from utils.appliance import Navigatable
+from utils.appliance.endpoints.ui import CFMENavigateStep, navigator, navigate_to
+from . import pol_btn
 
 list_tbl = CheckboxTable(table_locator="//div[@id='list_grid']//table")
 
-nav.add_branch(
-    'containers_image_registries',
-    {
-        'containers_image_registry':
-        lambda ctx: list_tbl.select_row_by_cells(
-            {'Host': ctx['image_registry'].host, 'Provider': ctx['image_registry'].provider.name}),
 
-        'containers_image_registry_detail':
-        lambda ctx: list_tbl.click_row_by_cells(
-            {'Host': ctx['image_registry'].host, 'Provider': ctx['image_registry'].provider.name}),
-    }
-)
+def match_title_and_controller():
+    title_match = browser_title() == 'Image Registries'
+    controller_match = controller_name() == 'container_image_registry'
+    return title_match and controller_match
 
 
-class ImageRegistry(Taggable, SummaryMixin):
+class ImageRegistry(Taggable, SummaryMixin, Navigatable):
 
     def __init__(self, host, provider):
         self.host = host
         self.provider = provider
+        Navigatable.__init__(self)
 
     def load_details(self, refresh=False):
-        if not self._on_detail_page():
-            self.navigate(detail=True)
-        elif refresh:
+        navigate_to(self, 'Details')
+        if refresh:
             tb.refresh()
 
-    def navigate(self, detail=True):
-        if detail is True:
-            if not self._on_detail_page():
-                sel.force_navigate('containers_image_registry_detail',
-                    context={'image_registry': self})
-        else:
-            sel.force_navigate('containers_image_registry',
-                context={'image_registry': self})
+
+@navigator.register(ImageRegistry, 'All')
+class ImageRegistryAll(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Compute', 'Containers', 'Image Registries')(None)
+
+    def resetter(self):
+        tb.select('Grid View')
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+@navigator.register(ImageRegistry, 'Details')
+class ImageRegistryDetails(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        tb.select('List View')
+        list_tbl.click_row_by_cells({'Name': self.obj.name})
+
+
+@navigator.register(ImageRegistry, 'EditTags')
+class ImageRegistryEditTags(CFMENavigateStep):
+    prerequisite = NavigateToSibling('Details')
+
+    def am_i_here(self):
+        header_match = sel.is_displayed('//h3[contains(text(), "Tag Assignment")]')
```

Comment:
Does this page header include the image registry name? This would match on _any_ image registry edit tags page, not just the specific instance of an ImageRegistry.